docs/troubleshooting/gitlab-sast.md

```diff
@@ -0,0 +1,68 @@
+# Troubleshooting GitLab SAST
+
+GitLab includes a SAST analyzer that uses Semgrep to find vulnerabilities.
+If you're having trouble and you suspect the issue lies with Semgrep, you can find some advice below.
+If you suspect your issue lies with GitLab,
+check [GitLab's own SAST troubleshooting guide](https://docs.gitlab.com/ee/user/application_security/sast/#troubleshooting).
+
+[TOC]
+
+# The Semgrep SAST CI job is slow
+
+The Semgrep SAST job should take less than a minute
+to scan a large project with 50k lines of Python and TypeScript code.
+If you see worse performance,
+please [reach out](../support.md) to the Semgrep maintainers for help with tracking down the cause.
+Long runtimes are typically caused by just one rule or source code file taking too long.
+
+If you suspect large and complex source code files (such as minified JS or generated code)
+are making the job last too long, you might want to exclude these files from scanning.
+[GitLab's path exclusion feature](https://docs.gitlab.com/ee/user/application_security/sast/#vulnerability-filters)
+does not skip scanning excluded files. It scans everything, and then hides results from excluded files.
+
+To improve performance by 10x on a typical project, you can use [Semgrep CI](../semgrep-ci.md) directly:
```

Comment:
Might want to also add a link to the sample GitLab CI config: (/sample-ci-configs/#gitlab-ci)